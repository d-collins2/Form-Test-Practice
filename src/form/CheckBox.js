import React from 'react';

const CheckBox = ({ checked, handleChange, label, labelFor }) => {
    return (
      <div className="form-check">
          <input
            className="form-check-input"
            name={labelFor}
            type="checkbox"
            checked={checked}
            id={labelFor}
            onChange={handleChange}/>
          <label className="form-check-label">{label}</label>
      </div>
    )
}

export default CheckBox;
