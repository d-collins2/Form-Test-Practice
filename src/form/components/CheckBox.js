import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap'

const CheckBox = ({ checked, handleChange, label, name }) => {
    return (
        <Form.Group>
            <Form.Check
                checked={checked}
                id={name}
                name={name}
                onChange={handleChange}
                type="checkbox"/>
            <Form.Label>{label}</Form.Label>
        </Form.Group>
    )
}

CheckBox.propTypes = {
    checked: PropTypes.bool,
    handleChange: PropTypes.func,
    name: PropTypes.string,
    label: PropTypes.string
};

CheckBox.defaultProps = {
    checked: false,
    formError: "Error",
    handleChange: function(){},
    label: "Label"
}

export default CheckBox;
