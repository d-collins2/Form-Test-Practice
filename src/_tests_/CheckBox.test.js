import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-bootstrap'
import { shallow } from 'enzyme';
import CheckBox from '../form/components/Checkbox.js';

describe('<CheckBuox />', function(){

    it('Should render correctly', function(){
        const component = renderer.create(<CheckBox />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should have a single input tag', function(){
        const component = shallow(<CheckBox />);
        expect(component.find(Form.Check).length).toBe(1);
    })

    it('Should have the correct checkbox label', function(){
        const label = "label";
        const component = shallow(<CheckBox label={label}/>);
        expect(component.find(Form.Label).text()).toEqual(label);
    });

    it('Should call changeHandler once onChange', function(){
        const mockHandler = jest.fn();
        const component = shallow(<CheckBox handleChange={mockHandler}/>);
        const input = component.find(Form.Check);
        input.checked = true;
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
})
