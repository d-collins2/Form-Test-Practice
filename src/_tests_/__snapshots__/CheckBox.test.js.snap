// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBuox /> Should render correctly 1`] = `
<div
  className="form-group"
>
  <div
    className="form-check"
  >
    <input
      className="form-check-input position-static"
      disabled={false}
      type="checkbox"
    />
  </div>
  <label
    className="form-label"
  />
</div>
`;
